jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -310,9 +317,17 @@ def _parse_as_spectrum1d(path):
             # FITS file.
             if _check_is_file(data):
                 try:
-                    data = Spectrum1D.read(data)
+                    if ext != 1 or transpose:
+                        with fits.open(data) as hdulist:
+                            data = _parse_as_spectrum1d(hdulist, ext, transpose)
```

Comment:
The way I understand this is that if `ext` is provided at all, the parser does not take advantage of the built-in specutils readable data formats. I guess this is okay if that is what you really want.

Reply:
Yes, that's correct - I don't believe `Spectrum1D.read` is set up to accept an extension argument, so we have to handle it here.